WeakAuras/BossMods.lua

```diff
@@ -0,0 +1,1508 @@
+    state.extend = extendTimer
+    if extendTimer ~= 0 then
+      state.autoHide = true
+    end
+    state.paused = bar.paused
+    state.remaining = bar.remaining
+    state.isCooldown = bar.isCooldown
+  end,
+
+  TimerMatches = function(self, timerId, message, operator, spellId, counter, cast, cooldown)
+    if not self.bars[timerId] then
+      return false
+    end
+
+    local v = self.bars[timerId]
+    if spellId and spellId ~= "" and spellId ~= v.spellId then
+      return false
+    end
+    if message and message ~= "" and operator then
+      if operator == "==" then
+        if v.text ~= message then
+          return false
+        end
+      elseif operator == "find('%s')" then
+        if v.text == nil or not v.text:find(message, 1, true) then
+          return false
+        end
+      elseif operator == "match('%s')" then
+        if v.text == nil or not v.text:match(message) then
+          return false
+        end
+      end
+    end
+    if counter then
+      counter:SetCount(tonumber(v.count) or 0)
+      if not counter:Match() then
+        return false
+      end
+    end
+    if cast ~= nil and v.cast ~= cast then
+      return false
+    end
+    if cooldown ~= nil and v.isCooldown ~= cooldown then
+      return false
+    end
+    return true
+  end,
+
+  GetStage = function(self)
+    return self.currentStage
+  end,
+
+  GetAllTimers = function(self)
+    return self.bars
+  end,
+
+  GetTimerById = function(self, timerId)
+    return self.bars[timerId]
+  end,
+
+  GetTimer = function(self, text, operator, spellId, extendTimer, counter, cast)
+    local bestMatch
+    for timerId, bar in pairs(self.bars) do
+      if self:TimerMatches(timerId, text, operator, spellId, counter, cast)
+      and (bestMatch == nil or bar.expirationTime < bestMatch.expirationTime)
+      and bar.expirationTime + extendTimer > GetTime()
+      then
+        bestMatch = bar
+      end
+    end
+    return bestMatch
+  end,
+
+  RecheckTimers = function(self)
+    local now = GetTime()
+    self.nextExpire = nil
+    for timerId, bar in pairs(self.bars) do
+      if not bar.paused then
+        if bar.expirationTime < now then
+          self.bars[timerId] = nil
+          WeakAuras.ScanEvents("BigWigs_StopBar", timerId)
+          if self.isGeneric then
+            WeakAuras.ScanEvents("BossMod_TimerStop", timerId)
+          end
+        elseif self.nextExpire == nil then
+          self.nextExpire = bar.expirationTime
+        elseif bar.expirationTime < self.nextExpire then
+          self.nextExpire = bar.expirationTime
+        end
+      end
+    end
+
+    if self.nextExpire then
+      self.recheckTimer = timer:ScheduleTimerFixed(self.RecheckTimers, self.nextExpire - now, self)
+    end
+  end,
+
+  EventCallback = function(self, event, ...)
+    if event == "BigWigs_Message" then
+      WeakAuras.ScanEvents("BigWigs_Message", ...)
+      if self.isGeneric then
+        WeakAuras.ScanEvents("BossMod_Announce", ...)
```

Comment:
Boss Mod Announce trigger expects message and icon as first and second arg of BossMod_Announce (as in DBM_Announce), but BigWigs_Message has them as third and fifth arg, respectively.